### ISSUE TITLE
Sampling: incremental printing of generated text

### ISSUE THREAD

**Issue by gwern** [jcjohnson/torch-rnn#71]
While using `sample.lua` to generate text to inspect how the quality is changing between checkpoints, it would be nice if each character could be printed as it's generated (like in `char-rnn`) so you don't have to wait for the full sample to be created before you see any output. This wait is particularly painful if you need to generate large amounts of text to see how it's going; for my metadata training, I need 10-20k characters to see it sample from several different authors, and it takes something like an hour to do that much.

`sample.lua` calls `LanguageModel.lua`, and inside `sample`, I think this could be done by adding a print call using the `next_char` variable?

**Comment by AlekzNet** [comment]
Have a look at this PR: https://github.com/jcjohnson/torch-rnn/pull/8

**Comment by danindiana** [comment]
I agree that the output being shown is in some cases desirable and a user-friendly aspect of char-rnn. Even more troubling, however is that even when I attempt to sample the output using a modifier like | tee when writing a file torch-rnn doesn't show the output. Maybe this is a bug related to #158 ?
